src/test/ui/async-await/in-trait/async-associated-types2.rs

```diff
@@ -0,0 +1,32 @@
+// edition: 2021
+
+#![feature(async_fn_in_trait)]
+#![allow(incomplete_features)]
+
+use std::future::Future;
+
+trait MyTrait {
+    type Fut<'a>: Future<Output = i32>
+    where
+        Self: 'a;
+
+    fn foo(&self) -> Self::Fut<'a>;
```

Comment:
Same as above, this isn't the error we're looking for here -- can you fix the lifetime?